sw/device/lib/base/macros.h

```diff
@@ -523,4 +523,17 @@ class SignConverter {
 #endif  // !defined(__ASSEMBLER__) && !defined(NOSTDINC) &&
         // !defined(RUST_PREPROCESSOR_EMIT)
 
+// This routine is needed to make sure that an interrupt does not sneak in
```

Comment:
Instead of describing how the macro works, I would describe what it achieves:
This routine waits until the condition `_irq` is true and it calls `wait_for_interrupt` in-between checks.
It gracefully handles race conditions between doing the check and waiting for an interrupt that might render the condition true.